pkl-commons-cli/src/main/kotlin/org/pkl/commons/cli/CliBaseOptions.kt

```diff
@@ -167,4 +168,26 @@ data class CliBaseOptions(
     projectDir?.resolve(ProjectDependenciesManager.PKL_PROJECT_FILENAME)
       ?: normalizedWorkingDir.getProjectFile(rootDir)
   }
+
+  /** [caCertificates] after normalization. */
+  val normalizedCaCertificates: List<Path> = caCertificates.map(normalizedWorkingDir::resolve)
+
+  /**
+   * The HTTP client shared between CLI commands created with this [CliBaseOptions] instance.
+   *
+   * To release the resources held by the HTTP client in a timely manner, call its `close()` method.
```

Comment:
I didn't change `CliBaseOptions` to implement `AutoCloseable`. Usually, an options object shouldn't need to be closed. Adding `CliBaseOptions.httpClient` felt a bit smelly, but it seemed like a price worth paying for being able to share `HttpClient` resources between commands. (I couldn't find another way without a major redesign.) I documented calling `CliBaseOptions.httpClient.close()` for the edge cases where it might matter.

Let me know if you want me to change `CliBaseOptions` to implement `AutoCloseable`. Kotlin doesn't have try-with-resources, but it does have a `Closeable.use` extension method.

Reply:
> Probably makes more sense to either accept the options on HttpClient.Builder directly, or accept HttpClient.Builder as an option as a whole.

Sorry I don't understand what you're proposing. Can you elaborate?
I considered passing `HttpClient` to `CliCommand`, but then `CliBaseOptions.caCertificates` is no longer meaningful. (It's `HttpClient` that needs to be configured with the certificates.)

Reply:
Sorry, so, one suggestion is to add the builder methods on `HttpClient.Builder` to `EvaluatorBuilder`:

```java
class EvaluatorBuilder {
  public EvaluatorBuilder addCertificates(Path file) { /* etc */ }

  public EvaluatorBuilder addCertificates(URI file) { /* etc */ }
```

Another option is to accept an `HttpClient.Builder` as an option in `EvaluatorBuilder:

```java
class EvaluatorBuilder {
  public HttpClient.Builder setHttpClientBuilder(HttpClient.Builder builder) { /* etc */ }
}
```

Reply:
Currently we have `EvaluatorBuilder.setHttpClient()`. Your proposals might make EvaluatorBuilder more convenient to use if the default HTTP client isn't good enough. But I don't see how any of this relates to the `CliBaseOptions.httpClient` discussion, and it's not clear to me if and what change you want there.